File: library/src/main/java/com/kimcy929/iconpakagereader/iconhelper/IconPackManager.java
package com.kimcy929.iconpakagereader.iconhelper;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by kimcy929 08/11/2016
 */
public class IconPackManager {

    private final static String[] sSupportedActions = new String[]{
            "org.adw.launcher.THEMES",
            "com.gau.go.launcherex.theme"
    };

    private static final String[] sSupportedCategories = new String[]{
            "com.fede.launcher.THEME_ICONPACK",
            "com.anddoes.launcher.THEME",
            "com.teslacoilsw.launcher.THEME"
    };

    public static Map<String, IconPackInfo> getSupportedPackages(Context context) {
        Intent iconIntent = new Intent();
        Map<String, IconPackInfo> packages = new HashMap<>();
        PackageManager packageManager = context.getPackageManager();
        for (String action : sSupportedActions) {
            iconIntent.setAction(action);
            for (ResolveInfo r : packageManager.queryIntentActivities(iconIntent, 0)) {
                IconPackInfo info = new IconPackInfo(r, packageManager);
                packages.put(r.activityInfo.packageName, info);
            }
        }
        iconIntent = new Intent(Intent.ACTION_MAIN);
        for (String category : sSupportedCategories) {
            iconIntent.addCategory(category);
            for (ResolveInfo r : packageManager.queryIntentActivities(iconIntent, 0)) {
                IconPackInfo info = new IconPackInfo(r, packageManager);
                packages.put(r.activityInfo.packageName, info);
            }
            iconIntent.removeCategory(category);
        }
        return packages;
    }
}

File: library/src/main/java/com/kimcy929/iconpakagereader/iconhelper/IconPackManager.kt
package com.kimcy929.iconpakagereader.iconhelper

import android.content.Context
import android.content.Intent
import java.util.*

/**
 * Created by kimcy929 08/11/2016
 */
object IconPackManager {

    private val sSupportedActions = arrayOf("org.adw.launcher.THEMES", "com.gau.go.launcherex.theme")

    private val sSupportedCategories =
        arrayOf("com.fede.launcher.THEME_ICONPACK", "com.anddoes.launcher.THEME", "com.teslacoilsw.launcher.THEME")

    fun getSupportedPackages(context: Context): Map<String, IconPackInfo> {
        var iconIntent = Intent()
        val packages = HashMap<String, IconPackInfo>()
        val packageManager = context.packageManager
        for (action in sSupportedActions) {
            iconIntent.action = action
            for (r in packageManager.queryIntentActivities(iconIntent, 0)) {
                val info = IconPackInfo(r, packageManager)
                packages[r.activityInfo.packageName] = info
            }
        }
        iconIntent = Intent(Intent.ACTION_MAIN)
        for (category in sSupportedCategories) {
            iconIntent.addCategory(category)
            for (r in packageManager.queryIntentActivities(iconIntent, 0)) {
                val info = IconPackInfo(r, packageManager)
                packages[r.activityInfo.packageName] = info
            }
            iconIntent.removeCategory(category)
        }
        return packages
    }
}
